### DISCUSSION TITLE
[@types/supertest] `get` return type is wrong

### DISCUSSION THREAD

**Discussion by erunion** [DefinitelyTyped/DefinitelyTyped#63864] [Issues with a @types package]
I'm not sure if this is a typing issue in `supertest` or `superagent` but I'm encountering it while using `supertest` with the following code:

```js
const request = require('supertest');

await request
  .agent(hub /* an instance of one of our request agents */)
  .set('Host', `${parentProject.subdomain}.${config.get('domain')}`)
  .get('/')
  .expect(200);
```

```
Property 'expect' does not exist on type 'string'. ts(2339)
```

Looks like the type of `get` in `superagent`, as of v4.1.16, is `get(field: string): string;`. I believe this should actually be typed as `this` instead.

[a comment by an automated account is not shown]

**Comment by aqemi** [comment]
I think that the issue is inside superagent typings.
`.set()` method returns `superagent.Request` interface which doesn't include HTTP call methods.

The following .d.ts file helped me to fix usage with supertest
```ts
declare module 'superagent' {
  interface Request {
    set(key: string, val: string): SuperAgentTest;
  }
}

```
You can use a different return type instead SuperAgentTest if you plan to use only superagent.
